bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieNettyServer.java

```diff
@@ -219,7 +225,12 @@ protected void initChannel(SocketChannel ch) throws Exception {
             });
 
             // Bind and start to accept incoming connections
-            bootstrap.bind(address.getAddress(), address.getPort()).sync();
+            Channel listen = bootstrap.bind(address.getAddress(), address.getPort()).sync().channel();
+            if (listen.localAddress() instanceof InetSocketAddress) {
+                if (conf.getBookiePort() == 0) {
+                    conf.setBookiePort(((InetSocketAddress) listen.localAddress()).getPort());
```

Comment:
@jiazhai @sijie is there any other case in which we "change" the provided configuration object ?
wouldn't it be clearer to add a public getBookieSocketAddress() ?
if we do not have any other case it should be better to treat ServerConfiguration as immutable once the server is started